micmac_li3ds/api.py

```diff
@@ -402,15 +402,15 @@ def update_obj(args, metadata, obj, type_):
         obj.setdefault('description', 'Imported from "{basename}"')
     if type_ in args:
         obj.update({k: v for k, v in args[type_].items() if v})
-    for key in obj:
-        if obj[key]:
+    for key in list(obj.keys()):
+        if obj[key] and isinstance(obj[key], str):
             try:
                 obj[key] = obj[key].format(**metadata)
-            except AttributeError:
-                continue
             except KeyError as e:
-                err = 'metadata {} not available for {}/{}="{}"'
-                raise KeyError(err.format(e, type_, key, obj[key]))
+                # obj[key] contain replacements fields that have no
+                # corresponding keys in metadata, just delete the
+                # key from the object in that case
+                del obj[key]
```

Comment:
I like this suggestion. I did this may, because I did not want to keep in `metadata` the keys with `None` values. In `import_orimatis` metadata is also passed to `sensor_camera` and `referential_world`, and I thought that it would be safer to pass a "clean" version of `metadata` to these functions.